src/formats/simple.hpp

```diff
@@ -0,0 +1,157 @@
+	        return false;
+	    }
+	    const simple_hdr_type* pkt_hdr  = (simple_hdr_type*)pkt_ptr;
+	    const uint8_t*        pkt_pld  = pkt_ptr  + sizeof(simple_hdr_type);
+	    int                   pld_size = pkt_size - sizeof(simple_hdr_type);
+	    pkt->seq =   be32toh(pkt_hdr->seq);
+            pkt->nsrc = 1;
+            pkt->src = 0;
+	    pkt->payload_size = pld_size;
+	    pkt->payload_ptr  = pkt_pld;
+	    pkt->nchan =         1;
+	    pkt->chan0 =   0;
+	    return this->valid_packet(pkt);
+    }
+};
+
+class SIMPLEProcessor : virtual public PacketProcessor {
+public:
+    inline void operator()(const PacketDesc* pkt,
+                           uint64_t          seq0,
+                           uint64_t          nseq_per_obuf,
+                           int               nbuf,
+                           uint8_t*          obufs[],
+                           size_t            ngood_bytes[],
+                           size_t*           src_ngood_bytes[]) {
+	    int    obuf_idx = ((pkt->seq - seq0 >= 1*nseq_per_obuf) +
+	                       (pkt->seq - seq0 >= 2*nseq_per_obuf));
+	    size_t obuf_seq0 = seq0 + obuf_idx*nseq_per_obuf;
+	    size_t nbyte = pkt->payload_size * BF_UNPACK_FACTOR;
+	    ngood_bytes[obuf_idx]               += nbyte;
+	    src_ngood_bytes[obuf_idx][0] += nbyte;
+	    // **CHANGED RECENTLY
+	    int payload_size = pkt->payload_size;//pkt->nchan*(PKT_NINPUT*2*PKT_NBIT/8);
+	
+	    size_t obuf_offset = (pkt->seq-obuf_seq0)*pkt->nsrc*payload_size;
+	    typedef unaligned256_type itype;
+	    typedef aligned256_type otype;
+	
+	    obuf_offset *= BF_UNPACK_FACTOR;
+	
+	    // Note: Using these SSE types allows the compiler to use SSE instructions
+	    //         However, they require aligned memory (otherwise segfault)
+	    itype const* __restrict__ in  = (itype const*)pkt->payload_ptr;
+	    otype*       __restrict__ out = (otype*      )&obufs[obuf_idx][obuf_offset];
+	
+	    int chan = 0;
+			//cout << pkt->src << ", " << pkt->nsrc << endl;
+		  //cout << pkt->nchan << endl;
+			for( ; chan<pkt->nchan; ++chan ) {
+#if defined BF_AVX_ENABLED && BF_AVX_ENABLED
+           _mm256_store_si256(reinterpret_cast<__m256i*>(&out[pkt->src + pkt->nsrc*chan]),
+					                    _mm256_loadu_si256(reinterpret_cast<const __m256i*>(&in[chan])));
+#else
+#if defined BF_SSE_ENABLED && BF_SSE_ENABLED
+           const unaligned128_type* dsrc = (const unaligned128_type*) &in[chan];
+           aligned128_type* ddst = (aligned128_type*) &out[pkt->src + pkt->nsrc*chan];
+           
+           _mm_store_si128(reinterpret_cast<__m128i*>(ddst),
+					                 _mm_loadu_si128(reinterpret_cast<const __m128i*>(dsrc)));
+           _mm_store_si128(reinterpret_cast<__m128i*>(ddst+1),
+					                 _mm_loadu_si128(reinterpret_cast<const __m128i*>(dsrc+1)));
+#else
+						::memcpy(&out[pkt->src + pkt->nsrc*chan],
+						      	 &in[chan], sizeof(otype));
+#endif
+#endif
```

Comment:
Since we are setting `nchan` to 1 in `SIMPLEDecoder::operator()` this loop would only copy the first 32B out of a packet and into the ring buffer?